src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -705,33 +711,74 @@ const WorkflowView = _.flow(
               div([
                 div({ style: { height: '2rem', fontWeight: 'bold' } }, ['Step 1']),
                 label(['Select root entity type:']),
-                h(Select, {
+                h(GroupedSelect, {
                   'aria-label': 'Entity type selector',
                   isClearable: false,
                   isDisabled: currentSnapRedacted || this.isSingle() || !!Utils.editWorkspaceError(ws),
                   isSearchable: true,
                   placeholder: 'Select data type...',
                   styles: { container: old => ({ ...old, display: 'inline-block', width: 200, marginLeft: '0.5rem' }) },
                   value: selectedEntityType,
-                  onChange: selection => {
-                    const value = this.updateEntityType(selection)
-                    this.setState({ entitySelectionModel: this.resetSelectionModel(value) })
+                  onChange: async ({ value, source }) => {
+                    if (source === 'snapshot') {
+                      const snapshotId = _.find({ name: value }, availableSnapshots).reference.snapshot
+                      const { tables } = await Ajax(signal).DataRepo.snapshot(snapshotId).details()
```

Comment:
1. could you cache the result of this somewhere, so we don't query for the same data every time if a user swaps the root entity type selection?
2. consider calling Rawls' entity type metadata endpoint, specifying the snapshot reference name, instead of calling data repo directly. It's the same data, but Rawls is already parsing it out for you. And if in the future Rawls supports entity types that are not 1:1 with table names - for instance, if Rawls supports an entity type that is the result of a join - you'd automatically inherit that here by calling Rawls instead of DR.

Reply:
further - if you called Rawls' entity type metadata endpoint instead, you'd also get all the attribute names for each entity type, which would allow you to satisfy the typeahead&validation when editing the method config inputs/outputs